tools/shell/linenoise.cpp

```diff
@@ -945,13 +948,12 @@ void linenoiseEditHistoryNext(struct linenoiseState* l, int dir) {
 /* Delete the character at the right of the cursor without altering the cursor
  * position. Basically this is what happens with the "Delete" keyboard key. */
 void linenoiseEditDelete(struct linenoiseState* l) {
-    if (l->len > 0 && l->pos < l->chars) {
+    if (l->len > 0 && l->pos < l->len) {
         uint32_t newPos = nextChar(l);
         uint32_t oldPos = getBufCharPos(l);
         uint32_t charSize = newPos - oldPos;
         memmove(l->buf + oldPos, l->buf + newPos, l->len - oldPos);
         l->len -= charSize;
-        l->chars--;
```

Comment:
@anuchak We should have a mechanism to check whether the character to delete is a utf-8 character or not.

Reply:
I've changed the name of this variable to `totalUTF8Chars` and the mechanism is simple, just check if total size of the character > 1.

Also I'm not sure if this variable is even required or not, because I checked DuckDB's codebase and they have no such tracking variable, don't think it is required at all.